src/H5PLpath.c

```diff
@@ -801,8 +801,8 @@ H5PL__find_plugin_in_path_table(const H5PL_search_params_t *search_params, hbool
 
         /* Search for the plugin in this path */
         if (H5PL__find_plugin_in_path(search_params, found, H5PL_paths_g[u], plugin_info) < 0)
-            HGOTO_ERROR(H5E_PLUGIN, H5E_CANTGET, FAIL, "search in path %s encountered an error",
-                        H5PL_paths_g[u])
+            HDONE_ERROR(H5E_PLUGIN, H5E_CANTGET, FAIL, "search in path %s encountered an error",
+                        H5PL_paths_g[u]);
```

Comment:
With this change, you may want to consider clearing the error stack and setting `ret_value` to `SUCCEED` if the plugin gets found. Otherwise, I believe a missing directory will still fail this function and also print out the error stack for previous errors.

Reply:
Right, added test change and fix.